service/resource/vaultcrt/delete.go

```diff
@@ -0,0 +1,62 @@
+package vaultcrt
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apismetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+
+	"github.com/giantswarm/cert-operator/service/key"
+)
+
+func (r *Resource) GetDeleteState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	currentSecret, err := toSecret(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.Log("cluster", key.ClusterID(customObject), "debug", "finding out if the secret has to be deleted")
+
+	var secretToDelete *apiv1.Secret
+	if currentSecret != nil {
+		secretToDelete = currentSecret
+	}
+
+	r.logger.Log("cluster", key.ClusterID(customObject), "debug", "found out if the secret has to be deleted")
```

Comment:
I think we can reduce one log by just saying that we found out secret has to be be deleted or is deleted already.

Reply:
I think we can reduce logging in general. I am personally still in the middle of learning how to implement resources and which information are important to log. For now I would like to keep the scheme I used in all of my implementations and clean them up as soon as I know how it should be. Making a little change here and later there will only cause snowflaky logging and this would make me super sad.